src/algebra/group_power/lemmas.lean

```diff
@@ -49,13 +50,25 @@ theorem list.sum_repeat : ∀ (a : A) (n : ℕ), (list.repeat a n).sum = n •
 @[simp, norm_cast] lemma units.coe_pow (u : units M) (n : ℕ) : ((u ^ n : units M) : M) = u ^ n :=
 (units.coe_hom M).map_pow u n
 
+/-- If `x ^ n.succ = 1` then `x` has an inverse, `x^n` -/
+def invertible_of_pow_succ_eq_one (x : M) (n : ℕ) (hx : x ^ n.succ = 1) :
+  invertible x :=
+⟨x ^ n, (pow_succ' x n).symm.trans hx, (pow_succ x n).symm.trans hx⟩
+
+/-- If `x ^ n = 1` then `x` has an inverse, `x^(n - 1)` -/
```

Comment:
Is there a guideline or rule of thumb for that says that these arguments should all be explicit? (I think I would've made both `x` and `n` implicit, but it looks like things are different in this file.)
```suggestion
/-- If `x ^ n.succ = 1` then `x` has an inverse, `x^n`. -/
def invertible_of_pow_succ_eq_one (x : M) (n : ℕ) (hx : x ^ n.succ = 1) :
  invertible x :=
⟨x ^ n, (pow_succ' x n).symm.trans hx, (pow_succ x n).symm.trans hx⟩

/-- If `x ^ n = 1` then `x` has an inverse, `x^(n - 1)`. -/
```

Reply:
Yeah, I thought this was odd too.